apps/src/lib/node/ledger/ethereum_node/oracle/mod.rs

```diff
@@ -147,7 +165,22 @@ pub fn run_oracle(
 ///
 /// It also checks that once the specified number of confirmations
 /// is reached, an event is forwarded to the ledger process
-async fn run_oracle_aux(oracle: Oracle) {
+async fn run_oracle_aux(mut oracle: Oracle) {
+    tracing::info!("Oracle is awaiting initial configuration");
+    let config = match await_initial_configuration(&mut oracle.control).await {
```

Comment:
`config` is not a field on the `Oracle` because as we don't have an initial configuration when constructing it, such a field ends up being an `Option<Config>` (or equivalently some sort of dummy config that doesn't completely make sense in production, like we had before). Then in `process()`, we end up having to do things like `oracle.config.min_confirmations.unwrap()` even though we know by the point that we reach `process` that the oracle must have been configured.

Configuring the oracle after it has been constructed via a channel feels a bit overwrought, I think ideally the `Shell` would construct the `Oracle` with an initial `Config` as from storage (or not construct it at all if an oracle wasn't necessary) - and then we could just have a `config` field on the oracle that wasn't an `Option`.

Reply:
if we switch from the `futures::try_join!()` macro to `futures::future::try_join_all()`, we will end up with a more malleable API that will allow us to filter out sub-processes which aren't needed.
```rs
let oracle = None;
let sub_processes = vec![..., oracle].into_iter().flatten();
let results: Vec<Result<_, _>> = futures::future::try_join_all(sub_processes);
```
we could construct `Some` oracle future based on a config param, or return `None` otherwise